File: frontend-miniapp/miniprogram/features/trade/data/trade-api.ts
import { apiRequest } from '../../../core/http/api-client'
import type {
  AddCartItemRequest,
  Cart,
  CreateOrderRequest,
  Order,
  PaymentOrder,
} from '../domain/trade-models'

export const cartApi = {
  get(): Promise<Cart> {
    return apiRequest({ path: '/api/cart' })
  },

  addItem(input: AddCartItemRequest): Promise<void> {
    return apiRequest({ path: '/api/cart/items', method: 'POST', data: input })
  },

  updateQuantity(itemId: string, quantity: number): Promise<void> {
    return apiRequest({
      path: `/api/cart/items/${itemId}/quantity`,
      method: 'PUT',
      data: { quantity },
    })
  },

  updateChecked(itemId: string, checked: boolean): Promise<void> {
    return apiRequest({
      path: `/api/cart/items/${itemId}/checked`,
      method: 'PUT',
      // 后端契约：checked 为 0/1 整数，直接发布尔会反序列化失败
      data: { checked: checked ? 1 : 0 },
    })
  },

  removeItem(itemId: string): Promise<void> {
    return apiRequest({ path: `/api/cart/items/${itemId}`, method: 'DELETE' })
  },
}

export const orderApi = {
  create(input: CreateOrderRequest): Promise<Order[]> {
    return apiRequest({ path: '/api/orders', method: 'POST', data: input })
  },

  list(): Promise<Order[]> {
    return apiRequest({ path: '/api/orders' })
  },

  detail(orderId: string): Promise<Order> {
    return apiRequest({ path: `/api/orders/${orderId}` })
  },

  cancel(orderId: string): Promise<void> {
    return apiRequest({ path: `/api/orders/${orderId}/cancel`, method: 'PUT' })
  },

  confirm(orderId: string): Promise<void> {
    return apiRequest({ path: `/api/orders/${orderId}/confirm`, method: 'PUT' })
  },
}

export const paymentApi = {
  create(orderId: string): Promise<PaymentOrder> {
    return apiRequest({ path: '/api/payments', method: 'POST', data: { orderId } })
  },

  mockPay(paymentOrderId: string): Promise<PaymentOrder> {
    return apiRequest({ path: `/api/payments/${paymentOrderId}/mock-pay`, method: 'POST' })
  },

  detail(paymentOrderId: string): Promise<PaymentOrder> {
    return apiRequest({ path: `/api/payments/${paymentOrderId}` })
  },
}

export const refundApi = {
  create(orderId: string, reason?: string): Promise<void> {
    const data: Record<string, string> = { orderId }
    if (reason) data.reason = reason
    return apiRequest({ path: '/api/refunds', method: 'POST', data })
  },
}

File: frontend-app/lib/features/trade/data/trade_api.dart
import 'package:shopping_app/core/network/api_client.dart';
import 'package:shopping_app/core/network/api_response.dart';
import 'package:shopping_app/features/trade/domain/trade_models.dart';

class TradeApi {
  TradeApi(this._client);

  final ApiClient _client;

  Future<Cart> getCart() async {
    final response = await _client.get('/api/cart');
    return unwrapApiResponse(response.data, Cart.fromJson);
  }

  Future<void> addCartItem(AddCartItemRequest request) async {
    final response = await _client.post(
      '/api/cart/items',
      data: request.toJson(),
    );
    unwrapApiResponse<void>(response.data, (_) {});
  }

  Future<void> updateQuantity(String itemId, int quantity) async {
    final response = await _client.put(
      '/api/cart/items/$itemId/quantity',
      data: {'quantity': quantity},
    );
    unwrapApiResponse<void>(response.data, (_) {});
  }

  Future<void> updateChecked(String itemId, bool checked) async {
    final response = await _client.put(
      '/api/cart/items/$itemId/checked',
      data: {'checked': checked ? 1 : 0},
    );
    unwrapApiResponse<void>(response.data, (_) {});
  }

  Future<void> removeItem(String itemId) async {
    final response = await _client.delete('/api/cart/items/$itemId');
    unwrapApiResponse<void>(response.data, (_) {});
  }

  Future<List<Order>> createOrder({
    required String receiverName,
    required String receiverPhone,
    required String receiverAddress,
    String? remark,
  }) async {
    final response = await _client.post(
      '/api/orders',
      data: {
        'receiverName': receiverName,
        'receiverPhone': receiverPhone,
        'receiverAddress': receiverAddress,
        if (remark != null && remark.isNotEmpty) 'remark': remark,
      },
    );
    return unwrapApiResponse(
      response.data,
      (value) => readObjectList(value, Order.fromJson, message: '订单响应格式错误'),
    );
  }

  Future<List<Order>> listOrders() async {
    final response = await _client.get('/api/orders');
    return unwrapApiResponse(
      response.data,
      (value) => readObjectList(value, Order.fromJson, message: '订单列表格式错误'),
    );
  }

  Future<Order> orderDetail(String orderId) async {
    final response = await _client.get('/api/orders/$orderId');
    return unwrapApiResponse(response.data, Order.fromJson);
  }

  Future<void> cancelOrder(String orderId) async {
    final response = await _client.put('/api/orders/$orderId/cancel');
    unwrapApiResponse<void>(response.data, (_) {});
  }

  Future<void> confirmOrder(String orderId) async {
    final response = await _client.put('/api/orders/$orderId/confirm');
    unwrapApiResponse<void>(response.data, (_) {});
  }

  Future<PaymentOrder> createPayment(String orderId) async {
    final response = await _client.post(
      '/api/payments',
      data: {'orderId': orderId},
    );
    return unwrapApiResponse(response.data, PaymentOrder.fromJson);
  }

  Future<PaymentOrder> mockPay(String paymentOrderId) async {
    final response = await _client.post(
      '/api/payments/$paymentOrderId/mock-pay',
    );
    return unwrapApiResponse(response.data, PaymentOrder.fromJson);
  }

  Future<PaymentOrder> paymentDetail(String paymentOrderId) async {
    final response = await _client.get('/api/payments/$paymentOrderId');
    return unwrapApiResponse(response.data, PaymentOrder.fromJson);
  }

  Future<void> createRefund({
    required String orderId,
    double? amount,
    String? reason,
  }) async {
    final response = await _client.post(
      '/api/refunds',
      data: {
        'orderId': orderId,
        if (amount != null) 'amount': amount,
        if (reason != null && reason.isNotEmpty) 'reason': reason,
      },
    );
    unwrapApiResponse<void>(response.data, (_) {});
  }
}
